storage/src/common/storage_reference.cc

```diff
@@ -248,5 +261,42 @@ Future<Metadata> StorageReference::PutFileLastResult() {
 
 bool StorageReference::is_valid() const { return internal_ != nullptr; }
 
+Future<ListResult> StorageReference::ListAll() {
+  FIREBASE_ASSERT_RETURN(Future<ListResult>(), internal_->is_valid());
```

Comment:
Please change this to use the PIMPL pattern - call internal_'s List* methods, and put the stub code in all 3 versions of StorageReferenceInternal, instead of in the outer class.